tests/unit/src/kakarot/precompiles/test_ecadd.cairo

```diff
@@ -29,28 +31,117 @@ func test__ecadd_impl{
     // Given
     alloc_locals;
 
-    let input_word0 = Helpers.load_word(ECPOINT_BYTES_LEN, calldata);
-    let input_word1 = Helpers.load_word(ECPOINT_BYTES_LEN, calldata + ECPOINT_BYTES_LEN);
-    let input_word2 = Helpers.load_word(ECPOINT_BYTES_LEN, calldata + ECPOINT_BYTES_LEN * 2);
-    let input_word3 = Helpers.load_word(ECPOINT_BYTES_LEN, calldata + ECPOINT_BYTES_LEN * 3);
-    let x0: BigInt3 = Helpers.to_bigint(input_word0);
-    let y0: BigInt3 = Helpers.to_bigint(input_word1);
-    let x1: BigInt3 = Helpers.to_bigint(input_word2);
-    let y1: BigInt3 = Helpers.to_bigint(input_word3);
-
-    let point0 = EcPoint(x0, y0);
-    let point1 = EcPoint(x1, y1);
-    let (expected_point: EcPoint) = ec_add(point0, point1);
-    let expected_x = Helpers.bigint_to_felt(expected_point.x);
-    let expected_y = Helpers.bigint_to_felt(expected_point.y);
+    let x0: BigInt3 = Helpers.bytes32_to_bigint(calldata);
+    let y0: BigInt3 = Helpers.bytes32_to_bigint(calldata + G1POINT_BYTES_LEN);
+    let x1: BigInt3 = Helpers.bytes32_to_bigint(calldata + G1POINT_BYTES_LEN * 2);
+    let y1: BigInt3 = Helpers.bytes32_to_bigint(calldata + G1POINT_BYTES_LEN * 3);
 
     // When
-    let (output_len, output: felt*, gas_cost) = PrecompileEcAdd.run(calldata_len, calldata);
-    let output_x = Helpers.load_word(ECPOINT_BYTES_LEN, output);
-    let output_y = Helpers.load_word(ECPOINT_BYTES_LEN, output + ECPOINT_BYTES_LEN);
+    let point0: G1Point = G1Point(x0, y0);
+    let point1: G1Point = G1Point(x1, y1);
+    let (expected_point: G1Point) = ALT_BN128.ec_add(point0, point1);
+    let (bytes_expected_x_len, bytes_expected_result: felt*) = Helpers.bigint_to_bytes_array(
+        expected_point.x
+    );
+    let (bytes_expected_y_len, bytes_expected_y: felt*) = Helpers.bigint_to_bytes_array(
+        expected_point.y
+    );
+    // We fill `bytes_expected_result + bytes_expected_x_len` ptr with `bytes_expected_y` elements
+    Helpers.fill_array(
+        bytes_expected_y_len, bytes_expected_y, bytes_expected_result + bytes_expected_x_len
+    );
+    let (output_len, output: felt*, gas_used) = PrecompileEcAdd.run(calldata_len, calldata);
 
     // Then
-    assert expected_x = output_x;
-    assert expected_y = output_y;
+    TestHelpers.assert_array_equal(
+        array_0_len=bytes_expected_x_len + bytes_expected_y_len,
+        array_0=bytes_expected_result,
+        array_1_len=output_len,
+        array_1=output,
+    );
+
+    return ();
+}
+
+@external
+func test__ecadd_via_staticcall{
```

Comment:
so this could be done with a case in `test_cases.py`

Reply:
we can remove this right? as you've updated the `test_cases.py`

Reply:
Removed